### ISSUE TITLE
scorep/scalasca make broken in next

### ISSUE THREAD

**Issue by JosephThomasParker** [boutproject/BOUT-dev#998]
When building `next` with Scorep on Archer and gcc,
`./configure --with-scorep`
I get the following error at linking:
```
2fluid.o: file not recognized: File truncated
collect2: error: ld returned 1 exit status
ERROR: Execution failed: CC 2fluid.scorep_init.o 2fluid.o `/home/y07/y07/scalasca/packages/scorep/2.0.2-gnu/bin/scorep-config --thread=none --mpp=none --mutex=none --nocompiler --nocuda --noonline-access --user --noopencl --memory=c++L64,libc --constructor`  -lm -L../../../lib -lbout++ -lfftw3 -L/opt/cray/netcdf-hdf5parallel/4.4.1.1/GNU/5.1/lib -lnetcdf_c++4 -lnetcdf -L../../../lib -lpvode -lpvpre `/home/y07/y07/scalasca/packages/scorep/2.0.2-gnu/bin/scorep-config --thread=none --mpp=none --mutex=none --nocompiler --nocuda --noonline-access --user --noopencl --memory=c++L64,libc --ldflags` `/home/y07/y07/scalasca/packages/scorep/2.0.2-gnu/bin/scorep-config --thread=none --mpp=none --mutex=none --nocompiler --nocuda --noonline-access --user --noopencl --memory=c++L64,libc --libs`   -o 2fluid
make: *** [2fluid] Error 1
```

Git bisect narrows it down to #953 -- undoing the changes in `make.config.in` fixes the problem.  @dschwoerer, would it be possible to make these changes compatible with Scorep?

**Comment by bendudson** [comment]
May be related to issue #997, or at least the troublesome line was added in the same commit.

**Comment by dschwoerer** [comment]
Another solution would be to just remove the dependency tracking completely.

If it is an additional target it will still cause the same issues (if I understand it correctly)

I thought it would be a nice improvement - but if it is causing that much issues - I am happy to remove it again.

I would rather have faster merges, that get undone if they break, than have slow merges :smile:

**Comment by bendudson** [comment]
`make depends` could be an optional target; if the standard targets don't depend on it then the dependencies would not normally be built. If someone is going to be doing development of BOUT++, then they can run `make depends` to speed up build times.

**Comment by dschwoerer** [comment]
Maybe rather add at configure time:
`./configure --enable-header-change-tracking` - that way the header file tracking will always be up to date?

**Comment by ZedThree** [comment]
Solved in #1031
